File: core/src/main/java/org/ogee/osgi.java
package org.ogee;

import org.osgi.framework.BundleContext;
import org.osgi.framework.ServiceReference;

public class osgi {
	
	public static void registerService(BundleContext bc, String name, Object service) {
		bc.registerService(name, service, null);
	}
	
	public static Object getService(BundleContext bc, String name) {
		ServiceReference ref = bc.getServiceReference(name);
		return bc.getService(ref);
	}
	
}

File: core/src/main/clojure/ogee/osgi.clj
(ns ogee.osgi
	(:import (org.ogee osgi)))
	
(defn init-module
	[name-ns name-var context]
	(let [atm ((ns-interns (symbol name-ns)) (symbol name-var))]
		(if (not= atm nil)
			(reset! (deref atm) context))))
	
(defn service-export
	[context service]
	(org.ogee.osgi/registerService context "java.util.Map" service))
